hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -396,10 +405,7 @@ private void ensurePartitionsLoadedCorrectly(List<String> partitionList) {
           LOG.debug("Time taken to list partitions {} ={}", partitionSet, (endLsTs - beginLsTs));
           pathInfoMap.forEach((partitionPair, statuses) -> {
             String relativePartitionStr = partitionPair.getLeft();
-            List<HoodieFileGroup> groups = addFilesToView(relativePartitionStr, statuses);
-            if (groups.isEmpty()) {
-              storePartitionView(relativePartitionStr, Collections.emptyList());
-            }
```

Comment:
This code was repeated so I moved it into the `addFilesToView` method

Reply:
shoud we do this check in line 185?

Reply:
Not sure what you mean here. line 185 was using streams API so empty is a no-op. It is handled on line 204 in the new code. I did not return early so the code will still end the timer and log the debug details.

Reply:
Let's return early and log some different msgs.